torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -107,7 +108,7 @@ std::shared_ptr<NCCLComm> NCCLComm::create_scalable(
           commIds.data(),
           &config),
       std::nullopt);
-  comm->ncclId_ = commIds[rank];
+  comm->ncclId_ = commIds[rootIdx];
```

Comment:
As previously mentioned, the *first* NCCL ID matters more.
Also so that all ranks prints the same NCCL ID in their log when representing this comm.

Reply:
Are you saying that we should do `comm->ncclId_ = commIds[0];` instead?

Reply:
Yes, probably.